crates/sui-bridge-indexer/src/main.rs

```diff
@@ -82,52 +80,93 @@ async fn main() -> Result<()> {
             .interval(std::time::Duration::from_millis(2000)),
     );
     let bridge_address = EthAddress::from_str(&config.eth_sui_bridge_contract_address)?;
-    let sui_bridge = EthSuiBridge::new(bridge_address, provider.clone());
-    let committee_address: EthAddress = sui_bridge.committee().call().await?;
-    let limiter_address: EthAddress = sui_bridge.limiter().call().await?;
-    let vault_address: EthAddress = sui_bridge.vault().call().await?;
-    let committee = EthBridgeCommittee::new(committee_address, provider.clone());
-    let config_address: EthAddress = committee.config().call().await?;
 
     let eth_client = Arc::new(
         EthClient::<ethers::providers::Http>::new(
             &config.eth_rpc_url,
-            HashSet::from_iter(vec![
-                bridge_address,
-                committee_address,
-                config_address,
-                limiter_address,
-                vault_address,
-            ]),
+            HashSet::from_iter(vec![bridge_address]),
         )
         .await?,
     );
 
-    let contract_addresses = HashMap::from_iter(vec![(bridge_address, config.start_block)]);
+    let pg_pool = get_connection_pool(config.db_url.clone());
+
+    // capture finalized blocks
```

Comment:
don't know if it is possible but it would be really cool to abstract the finalized and unfinalized initialization in a function and clean up the `main` function which is getting difficult to think about. 
Maybe even move out the sui initialization?
Don't know thinking out loud here...